CONTRIBUTING.md

```diff
@@ -174,23 +174,29 @@ When submitting the PR remember to label it with the 📖 (:book:) icon.
 
 ## Releases
 
-Cluster API uses [GitHub milestones](https://github.com/kubernetes-sigs/cluster-api/milestones) to track releases. Issues in a release milestone have been prioritized and accepted for the release. However, these issues are not committed to the release, unless they are marked as `kind/release-blocking`. Getting them into the release is dependent on someone in the community getting assigned to the issue and completing the work.
-
 - Minor versions CAN be planned and scheduled for each quarter, or sooner if necessary.
   - Each minor version is preceded with one or more planning session.
   - Planning consists of one or more backlog grooming meetings, roadmap amendments,
     and CAEP proposal reviews.
-- Patch versions CAN be planned and scheduled each month for each of the currently supported series (usually N and N-1).
-- Code freeze is in effect 72 hours (3 days) before a release.
+  - Cluster API uses [GitHub milestones](https://github.com/kubernetes-sigs/cluster-api/milestones) to track work
+    for minor releases. 
+  - Adding an issue to a milestone provides forward visibility on what the next release will be, so, as soon as there
```

Comment:
In my opinion we shouldn't wait, except something has been explicitly defined as release-blocking

Reply:
I agree. In that case, it might be worth making this explicit so it's clear to contributors and users that an issue being in the milestone doesn't guarantee inclusion in the release.

Reply:
+1 to not wait

Reply:
Updated:
- Adding an issue to the release only imply the issue will be tracked
- We should add to milestone/track only major features/bug fixes specifically targeting a release; everything else will simply merge when ready without additional toil
- There is an item calling out that an issue being in the milestone doesn't guarantee inclusion in the release